pkgs/by-name/ca/cargo-tauri/hook.nix

```diff
@@ -0,0 +1,55 @@
+{
+  lib,
+  stdenv,
+  makeSetupHook,
+  cargo,
+  cargo-tauri,
+  rust,
+  # The subdirectory of `target/` from which to copy the build artifacts
+  targetSubdirectory ? stdenv.hostPlatform.rust.cargoShortTarget,
+}:
+
+let
+  kernelName = stdenv.hostPlatform.parsed.kernel.name;
+in
+makeSetupHook {
+  name = "tauri-hook";
+
+  propagatedBuildInputs = [
+    cargo
+    cargo-tauri
+  ];
+
+  substitutions = {
+    inherit targetSubdirectory;
+    inherit (rust.envVars) rustHostPlatformSpec setEnv;
+
+    # A map of the bundles used for Tauri's different supported platforms
+    # https://github.com/tauri-apps/tauri/blob/23a912bb84d7c6088301e1ffc59adfa8a799beab/README.md#platforms
+    defaultTauriBundleType =
+      {
+        darwin = "app";
+        linux = "deb";
+      }
+      .${kernelName} or (throw "${kernelName} is not supported by cargo-tauri.hook");
+
+    # $tmpDir is the path to the build artifacts (i.e., `./target/release`)
+    installScript =
+      {
+        darwin = ''
+          mv "$tmpDir"/bundle/macos $out/Applications
+        '';
+
+        linux = ''
+          mv "$tmpDir"/bundle/deb/*/data/usr $out
```

Comment:
I don't think there should be any cases where this can happen, but yeah I agree this is a big footgun. As the single directory (currently) matched by this glob is named after the application though (which may be different from our package's `pname`, etc), I'm not sure how else to go about this

Maybe we could try to figure out the name from the Tauri configuration?

Reply:
Now I think about it again: a better option would be to use `jq` on `tauri.conf`. We should reliably be able to find that file based on the current build directory (`buildAndTestSubdir`)

Reply:
TL;DR: After revisiting this, it seems we will need to keep it

Tauri's deb bundles output both a packaged .deb and an unpacked version to `$targetDir/bundle/deb`. The basename of both of these is in the format of `<productName>_<version>_<arch>`, with `<arch>` being determined by [this](https://github.com/tauri-apps/tauri/blob/58bab8b35b6e3d4ddcc229920081c878bd7c7fd6/crates/tauri-bundler/src/bundle/linux/debian.rs#L44-L48) match statement internally. We can grab the `productName` from `tauri.conf.json` (which should always be in the current build directory as said before) and figure out the architecture by reimplementing that same match statement; ***however***, we have no consistent way to find the version as that is read from Cargo.toml, which may have it under `package.version` or `workspace.package.version`. The best we could do is attempt to find it in either of these and fail/fallback to a variable if nothing is found, but IMO this is introducing a lot of complication for something that's still quite fragile

We should be fine though, as the only way for multiple directories to be matched here would be if we're building multiple targets, which this hook doesn't support in the first place